### PR TITLE
Users/dvoituron/autocomplete

### PR DESCRIPTION
**Pull request by dvoituron**
# New component **FluentAutocomplete**

Examples.

Autocomplete-Demo2

```xml
<FluentAutocomplete Id="my-autocomplete"
                    TOption="Country"
                    Width="250px"
                    Placeholder="Select countries"
                    OnOptionsSearch="@OnSearchAsync"
                    MaximumSelectedOptions="3"
                    OptionText="@(item => item.Name)"
                    @bind-SelectedOptions="@SelectedItems" />
```

```xml
<FluentAutocomplete Id="my-customized"
                    @ref="ContactList"
                    TOption="Person"
                    Width="100%"
                    Placeholder="search"
                    OnOptionsSearch="@OnSearch"
                    MaximumSelectedOptions="3"
                    OptionText="@(item => item.LastName)"
                    OptionStyle="min-height: 40px;"
                    @bind-SelectedOptions="@SelectedItems">

    @* Template used with each Selected items *@
    <SelectedOptionTemplate>
        <FluentOptionPeople ImageSize="24px"
                            Image="@context.Picture"
                            Name="@($"{context.FirstName} {context.LastName}")"
                            Style="height: 26px; background: var(--neutral-fill-secondary-hover)"
                            DismissTitle="Remove"
                            OnDismissClick="@(async () => await ContactList.RemoveSelectedItemAsync(context))" />
    </SelectedOptionTemplate>

    @* Template used with each Option items *@
    <OptionTemplate>
        <FluentOptionPeople ImageSize="32px"
                            Image="@context.Picture"
                            Name="@($"{context.FirstName} {context.LastName}")" />
    </OptionTemplate>

    <HeaderContent>
        <FluentLabel Color="Color.Accent"
                     Style="padding: 8px; font-size: 11px; border-bottom: 1px solid var(--neutral-fill-stealth-hover);">
            Suggested contacts
        </FluentLabel>
    </HeaderContent>

    <FooterContent>
        @if (!context.Any())
        {
            <FluentLabel Style="font-size: 11px; text-align: center; width: 200px;">
                No results found
            </FluentLabel>
        }
    </FooterContent>
</FluentAutocomplete>
```

This PR contains the component `FluentAutocmpleye<TOption>`, the component `FluentOptionPeople` and all associated Unit Tests.

image

### CODE REVIEW

File: src/Core/Components/List/FluentAutocomplete.razor [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                        {
+                            @SelectedOptionTemplate(item)
+                        }
+                    }
+                    &nbsp;
+                </fluent-horizontal-scroll>
+            }
+            <svg width="12" height="12" style="cursor: pointer;" xmlns="http://www.w3.org/2000/svg" slot="end" @onclick="@OnDropDownExpandedAsync">
+                <path d="M2.15 4.65c.2-.2.5-.2.7 0L6 7.79l3.15-3.14a.5.5 0 11.7.7l-3.5 3.5a.5.5 0 01-.7 0l-3.5-3.5a.5.5 0 010-.7z"></path>
+            </svg>
```
Comment (vnbaaij) [inline src/Core/Components/List/FluentAutocomplete.razor:59]: Do we not want to use an Icon here? Or is it not available in the Fluent UI System Icons set

File: src/Core/Components/List/FluentAutocomplete.razor [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        </fluent-text-field>
+
+        @* List of available items *@
+        @if (IsMultiSelectOpened)
```
Comment (vnbaaij) [inline src/Core/Components/List/FluentAutocomplete.razor:63]: We don't use `Is...`, right?

File: src/Core/Components/List/FluentAutocomplete.razor.cs [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+    /// </summary>
+    [Parameter]
+    public RenderFragment<IEnumerable<TOption>>? FooterContent { get; set; }
+
+    /// <summary />
+    private string? ListStyleValue => new StyleBuilder()
+                                                .AddStyle("width", Width, when: !string.IsNullOrEmpty(Width))
+                                                .AddStyle("display", "none", when: (Items == null || !Items.Any()) && 
+                                                                                   (HeaderContent != null || FooterContent != null))
+                                                .Build();
```
Comment (vnbaaij) [inline src/Core/Components/List/FluentAutocomplete.razor.cs:125]: Spacing is a bit weird

File: src/Core/Components/List/FluentAutocomplete.razor.cs [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+                return $"width: {Width}; min-width: {Width};";
+            }
+        }
+    }
+
+    /// <summary />
+    private string IdScroll => $"{Id}-scroll";
+
+    /// <summary />
+    private bool IsMultiSelectOpened { get; set; }
```
Comment (vnbaaij) [inline src/Core/Components/List/FluentAutocomplete.razor.cs:154]: Is... naming

**Review by vnbaaij — APPROVED** [review]

File: src/Core/Components/List/FluentAutocomplete.razor (hunk 1 above)
Comment (dvoituron) [reply src/Core/Components/List/FluentAutocomplete.razor:59]: Yes, that's the "Bottom arrow" icon at the end of the textfield, to open the popup.
May be better to include this code in `CoreIcon`. I will to that in a next PR.

**Pull request merged**